darshan-util/pydarshan/darshan/cli/summary.py

```diff
@@ -81,17 +83,28 @@ def get_encoded_fig(mpl_fig: Any):
         encoded_fig = base64.b64encode(tmpfile.getvalue()).decode("utf-8")
         return encoded_fig
 
-    def generate_img(self):
+    def generate_fig(self):
         """
-        Generate the image using the figure data.
+        Generate a figure using the figure data.
         """
-        # generate the matplotlib figure using the figure's
+        # generate the figure using the figure's
         # function and function arguments
-        mpl_fig = self.fig_func(**self.fig_args)
-        # encode the matplotlib figure
-        encoded = self.get_encoded_fig(mpl_fig=mpl_fig)
-        # create the img string
-        self.img_str = f"<img src=data:image/png;base64,{encoded} alt={self.fig_title} width={self.fig_width}>"
+        fig = self.fig_func(**self.fig_args)
+        supported_mpl_fig_types = [matplotlib.figure.Figure, seaborn.axisgrid.JointGrid]
+        if type(fig) in supported_mpl_fig_types:
+            # encode the matplotlib figure
+            encoded = self.get_encoded_fig(mpl_fig=fig)
+            # create the img string
+            self.fig_html = f"<img src=data:image/png;base64,{encoded} alt={self.fig_title} width={self.fig_width}>"
+        elif isinstance(fig, pd.DataFrame):
+            # use built-in pandas utility to convert table to html
+            self.fig_html = fig.to_html(index=False, border=0)
```

Comment:
I changed this from `header=False` to `index=False`, since that is what works best with the common access size table, but this obviously won't work for every case (i.e. the metadata and module tables use `header=False`). We will likely have to come up with a way to choose which one to remove, because so far the tables have always needed one of these turned off.

Reply:
Maybe leave this to the author of the table code for any given case? You could mandate a small custom class/object is returned for tables that gives you a method/attribute or similar that retrieves the html (which is expected to be formatted properly, otherwise it is up to the author to fix it).

```python
elif isinstance(fig, DarshanReportTable):
    # the function returning DarshanReportTable is
    # is responsible for formatting the html
    self.fig_html = DarshanReportTable.html
```

At the moment you could even get away with just providing the raw html without a custom class, but this may not be the best design if we plan on supporting more custom things later, like a class/object for traffic light heuristics, etc.

Reply:
Thanks, I like that idea. I'll see if I can implement that on the sibling PR along with your example.